backend/src/common/util/workflow.go

```diff
@@ -70,12 +70,19 @@ func NewWorkflowFromBytesJSON(bytes []byte) (*Workflow, error) {
 
 func NewWorkflowFromScheduleWorkflowSpecBytesJSON(bytes []byte) (*Workflow, error) {
 	var workflow workflowapi.Workflow
-	err := json.Unmarshal(bytes, &workflow.Spec)
+	err := json.Unmarshal(bytes, &workflow)
 	if err != nil {
-		return nil, NewInvalidInputErrorWithDetails(err, "Failed to unmarshal the inputs")
+		return nil, NewInvalidInputErrorWithDetails(err, "Failed to unmarshal the inputs into workflow")
+	}
+	if workflow.Spec.Entrypoint == "" {
```

Comment:
Yes, `entrypoint` is required. I tested without this fallback, old recurring runs will fail to schedule with Argo workflow controller error: 
> "invalid spec: spec.entrypoint is required" namespace=kubeflow workflow=runoftutorialdsl-controlss7xv6-2-2568324918"

Basically, `json.Unmarshal` tolerates for spec mismatch, it would just return an empty struct without any error. 
The other option I considered was to use `DisallowUnknownFields` from `json.Decoder`, but my concern was it might be too strict, so I landed on this "safer" change.